File: Back-end/gps/src/main/java/br/edu/ifpb/gps/gps/model/Visita.java
package br.edu.ifpb.gps.gps.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.sql.Date;
import java.time.LocalDate;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Visita {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    @ManyToOne
    Porteiro porteiro;
    @ManyToOne
    Residente residente;
    @OneToOne
    Visitante visitante;
    @DateTimeFormat(pattern = "dd/MM/yyyy")
    LocalDate data = LocalDate.now();
    boolean autorizado = false;

    public Visita(Porteiro porteiro, Residente residente, Visitante visitante){
        this.porteiro = porteiro;
        this.residente = residente;
        this.visitante = visitante;
        this.data = LocalDate.now();
    }
}

File: Front-end/material-dashboard-angular2-master/material-dashboard-angular2-master/src/app/interface/visita.ts
import { Porteiro } from "./porteiro";
import { Residente } from "./residente";
import { Visitante } from "./visitante";

export interface Visita {
    idVisita?: number,
    cadastro: String,
    autorizacao: String,
    visitanteId: String,
    data: Date
}
// visita{
//     id: null,
//     porteiro: idPorteiro,
//     autorizacao: idResidente,
//     visitante: idVisitante,
//     data: Date
// }